src/components/board/board.test.js

```diff
@@ -1,8 +1,42 @@
 
+const FILES = ['Jack', 'Seamore', 'Hugh', 'Michael', 'Benjamin', 'Dixie', 'Justin', 'William'];
+const RANKS = ['Dover', 'Normus', 'Butz', 'Jaenus', 'Kanoff', 'Sider', 'Stroker', 'unt'];
+const randomFiles = FILES[Math.floor(Math.random() * FILES.length)];
+const randomRanks = RANKS[Math.floor(Math.random() * RANKS.length)];
+const randomFileRank = `${randomRanks}${randomFiles}`;
 
+function makeBoard(ranks, files) {
+  return ranks.reduce((board, rank) => {
```

Comment:
You need to focus just testing the board component - not the makeBoard function. Mock the makeBoard function. Something like this:

```
import Board from '../board/board';
import boardService from '../../services/boardService';
import { RANKS, FILES } from '../../data.js';

const EXPECTED = [{ coordinate: 'A1' }];

describe ('Board Component', () => {
  let spyBoardService;
  beforeAll(() => {
    spyBoardService = jest.spyOn(boardService, 'makeBoard').mockImplementation(() => {
      return EXPECTED;
    });
  });
  beforeEach(() => {
    spyBoardService.mockClear();
  });
  afterAll(() => {
    spyBoardService.mockRestore();
  });

  test ('buildBoard returns an array of objects successfully', (done) => {
    const _board = Board();
    const squareId = JSON.stringify(_board).indexOf('A1');
    expect(spyBoardService).toHaveBeenCalledWith(FILES, RANKS);
    expect(squareId).not.toEqual(-1);
    done();
  });
});
```

Reply:
Even better i think you should really think about what this test is testing.

We've already tested the "makeBoard" function - though i argue you should actually export a completed board from that so you don't actually have to call "makeBoard" here - because it's brittle. An initial board is pretty much a static object. It should always be the same - that's a topic for the other test though.

Here i think you should be testing whether the data is displayed. Try using this as a starting kit:

```
import React from "react";
import TestRenderer from 'react-test-renderer'; // ES6
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import Board from "./board";
let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  // container *must* be attached to document so events work correctly.
  document.body.appendChild(container);
});
afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});
it("renders with or without a name", () => {
  let board = null;
  board = TestRenderer.create(<Board />);
  const boardAsJson = board.toJSON();
  expect(boardAsJson.props.className).toEqual('wrapper');
});
```

From there, inspect the boardAsJson and see if you can write some expectations for what you expect there.